server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,285 @@
+/**
+* The /fee_centers HTTP API endpoint
+*
+* @module finance/feeCenters
+*
+* @description This controller is responsible for implementing all crud and others custom request
+* on the fee_center table through the `/fee_centers` endpoint.
+*
+* @requires lib/db
+**/
+
+
+var db = require('../../lib/db');
+var NotFound = require('../../lib/errors/NotFound');
+
+/**
+* Returns an array of fee centers
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // GET /fee_centers : Get list of fee_centers
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.list(req, res, next);
+*/
+
+function list (req, res, next) {
+  'use strict';
+
+  var sql =
+    'SELECT fc.id, fc.label, fc.is_cost FROM fee_center AS fc ';
+  var filter = '', ordering = 'ORDER BY fc.label;';
+
+  if (req.query.full === '1') {
+    sql =
+      `
+        SELECT 
+	        fc.id, fc.label, fc.project_id, fc.is_cost, fc.note, fc.is_principal, p.name, p.abbr, p.enterprise_id, p.zs_id, tt.value
+        FROM
+	        fee_center AS fc
+        JOIN
+	        project AS p ON fc.project_id = p.id
+        LEFT JOIN
+	        (
+		        SELECT 
+			        IFNULL(IF(t.is_cost = 1, SUM(t.debit_equiv - t.credit_equiv), SUM(t.credit_equiv - t.debit_equiv)), 0) AS value, t.id 
+            FROM 
+			        (
+				        SELECT 
+					        gl.debit_equiv, gl.credit_equiv, f.is_cost, f.id 
+				        FROM 
+					        general_ledger AS gl 
+				        JOIN
+					        fee_center AS f ON gl.fc_id = f.id 
+			        ) AS t
+	        ) AS tt ON tt.id = fc.id
+      `;
+  }
+
+
+  if(req.query.available === '1') {
+    filter += ' fc.id NOT IN (SELECT s.cc_id FROM service AS s WHERE NOT ISNULL(s.cc_id)) AND fc.id NOT IN (SELECT s.pc_id FROM service AS s WHERE NOT ISNULL(s.pc_id)) ';
+  }
+
+  if(req.query.is_cost){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_cost = ${req.query.is_cost} `;
```

Comment:
What happens if `req.query.is_cost` is `(DELETE FROM general_ledger WHERE 1); -- ` ?